nixos/modules/services/mail/stalwart-mail.nix

```diff
@@ -53,6 +53,16 @@ in
         Data directory for stalwart
       '';
     };
+
+    secrets = lib.mkOption {
+      type = with lib.types; listOf path;
+      description = ''
+        A list of files containing the various secrets. Should be in the format
+        expected by systemd's `EnvironmentFile` directory. Secrets can be
+        referenced in the format `{env:VAR}`.
```

Comment:
Since stalwart mail also supports a `%{file:/path/to/file}%` macro to load values (the `%` signs should also be needed for env macros btw), I'd be happier if we could switch to a `LoadCredentials` based setup. That allows us both to automatically document which variables were declared and also let's the service manager to load the credentials for us so they can be owned solely by root.

So basically `type = attrsOf path` and a `LoadCredential = mapAttrsToList (n: v: "${n}:${v}") cfg.secrets`. This can then be accessed via `%{file:/run/credentials/stalwart-mail.service/VAR_NAME}%` (not sure if stalwart mail supports nested macros for `%{file:%{env:CREDENTIALS_DIRECTORY}%/VAR_NAME}%`.

Reply:
Thank you! Interesting, didn't know this format yet. Changed it but didn't test it yet